From fb013adc7d3c1cfefbd012c2d1ae4961f5a3a146 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Tue, 15 Aug 2023 09:08:08 +0000
Subject: [PATCH] udev: use bfq as the default scheduler

As requested in https://bugzilla.redhat.com/show_bug.cgi?id=1738828.
Test results are that bfq seems to behave better and more consistently on
typical hardware. The kernel does not have a configuration option to set
the default scheduler, and it currently needs to be set by userspace.

See the bug for more discussion and links.
---
 rules.d/60-block-scheduler.rules | 3 +++
 rules.d/meson.build              | 1 +
 2 files changed, 4 insertions(+)
 create mode 100644 rules.d/60-block-scheduler.rules

diff --git a/rules.d/60-block-scheduler.rules b/rules.d/60-block-scheduler.rules
new file mode 100644
index 0000000..30a4f4c
--- /dev/null
+++ b/rules.d/60-block-scheduler.rules
@@ -0,0 +1,3 @@
+ACTION=="add", SUBSYSTEM=="block", ENV{DEVTYPE}=="disk", \
+  KERNEL=="mmcblk*[0-9]|msblk*[0-9]|mspblk*[0-9]|sd*[!0-9]|sr*", \
+  ATTR{queue/scheduler}="bfq"
diff --git a/rules.d/meson.build b/rules.d/meson.build
index 20fca22..212c875 100644
--- a/rules.d/meson.build
+++ b/rules.d/meson.build
@@ -8,6 +8,7 @@ rules = [
         [files('60-autosuspend.rules',
                '60-block.rules',
                '60-cdrom_id.rules',
+               '60-block-scheduler.rules',
                '60-dmi-id.rules',
                '60-drm.rules',
                '60-evdev.rules',
-- 
2.39.0

